app/controllers/sessions_controller.rb

```diff
@@ -26,12 +26,20 @@ def create
 
   def twitter
     auth = request.env["omniauth.auth"]
-    user = User.where(uid: auth["uid"]).first || User.from_twitter(auth)
+
+    user = current_user
+
     if user
```

Comment:
You should be able to leave this as `if current_user` and not worry about creating that user var.  Aside from that, this is a great implementation.